Secret-agency-service/src/main/java/cz/fi/muni/pa165/secretagency/service/facade/AgentFacadeImpl.java

```diff
@@ -68,6 +75,30 @@ public AgentDTO getAgentByCodeName(String codename) {
         return beanMappingService.mapTo(agentService.getAgentByCodeName(codename), AgentDTO.class);
     }
 
+    @Override
+    public void assignAgentToMission(AgentDTO agent, MissionDTO mission) {
+        Mission mappedMission = beanMappingService.mapTo(missionService.getEntityById(mission.getId()), Mission.class);
+        Agent mappedAgent = beanMappingService.mapTo(agent, Agent.class);
+        mappedAgent.addMission(mappedMission);
+        agentService.save(mappedAgent);
+    }
+
+    @Override
+    public void removeAgentFromMission(AgentDTO agent, MissionDTO mission) {
+        Mission mappedMission = beanMappingService.mapTo(missionService.getEntityById(mission.getId()), Mission.class);
+        Agent mappedAgent = beanMappingService.mapTo(agent, Agent.class);
+        mappedAgent.removeMission(mappedMission);
```

Comment:
Stejně jako u addMission by se měla volat metoda na removeAgent z Mission. Aktorát nevím, jestli v současnosti existuje. 😄 
Je důležitý, aby při odebrání agenta z mise, byla daná mise odebrána u agenta a zároveň byl u mise odebrán agent. Předppkládám.